draft-ietf-ipsecme-ikev2-multiple-ke-00.xml

```diff
@@ -885,6 +885,129 @@ Type     Description                   Used In
     	&RFC8391;
     </references>
 
+  <section title="Sample Multiple Key Exchanges" anchor="sample-exchanges">
+    <t>This appendix shows some examples of multiple key exchanges.  These examples are purely for information
+    purposes and they describe some message flow scenarios that may occur in establishing an IKE or CHILD SA.  Note
+    that some payloads that are not relevant to multiple key exchanges may be omitted for brevity.
+    </t>
+
+    <section title="No Additional Key Exchange Used" anchor="sample-exchanges-none-selected">
+      <t>The initiator proposes two sets of optional additional key exchanges, but the responder does not
+      support any of them.  The responder chooses NONE for each set and consequently, IKE_INTERMEDIATE
+      exchange does not takes place and the exchange proceeds to IKE_AUTH phase.  The resulting keying
+      materials are the same as those derived with <xref target="RFC7296"/>.</t>
+
+      <figure><artwork align="center"><![CDATA[
+Initiator                     Responder
+------------------------------------------------------------------------
+HDR(IKE_SA_INIT), SAi1(.. AKE*...), --->
+KEi1, Ni, N(IKEV2_FRAG_SUPPORTED),
+N(INTERMEDIATE_EXCHANGE_SUPPORTED)
+  Proposal #1
+    Transform ECR (ID = ENCR_AES_GCM_16,
+                   256-bit key)
+    Transform PRF (ID = PRF_HMAC_SHA2_512)
+    Transform KE (ID = Curve25519)
+    Transform AKE1 (ID = PQ_KEM_1)
+    Transform AKE1 (ID = PQ_KEM_2)
+    Transform AKE1 (ID = NONE)
+    Transform AKE2 (ID = PQ_KEM_3)
+    Transform AKE2 (ID = PQ_KEM_4)
+    Transform AKE2 (ID = NONE)
+                         <--- HDR(IKE_SA_INIT), SAr1(.. AKE*...),
+                              KEr1, Nr, N(IKEV2_FRAG_SUPPORTED),
+                              N(INTERMEDIATE_EXCHANGE_SUPPORTED)
+                                Proposal #1
+                                  Transform ECR (ID = ENCR_AES_GCM_16,
+                                                 256-bit key)
+                                  Transform PRF (ID = PRF_HMAC_SHA2_512)
+                                  Transform KE (ID = Curve25519)
+                                  Transform AKE1 (ID = NONE)
+                                  Transform AKE2 (ID = NONE)
+
+HDR(IKE_AUTH), SK{ IDi, AUTH, SAi2, TSi, TSr } --->
+                         <--- HDR(IKE_AUTH), SK{ IDr, AUTH, SAr2,
+                              TSi, TSr }
+      ]]></artwork></figure>
+    </section>
+
+    <section title="Additional Key Exchange for CHILD SA only" anchor="sample-exchanges-ake-child-sas">
```

Comment:
I think that it's better to rename the section to "Additional Key Exchange in the CREATE_CHILD_SA Exchange Only". And I think it's better to show how to create fully QC-protected IKE SA, in this case so a clarification should be added that we rekey IKE SA (and not Child SA that was created in IKE_AUTH).

Reply:
Happy with that.